openapi/openapi.yaml

```
@@ -365,6 +432,110 @@ components:
           description: If decision logging is enabled, this field contains a string that uniquely identifies the decision. The identifier will be included in the decision log event for this decision. Callers can use the identifier for correlation purposes.
         provenance:
           $ref: "#/components/schemas/provenance"
+    CompileQuery:
+      type: string
+      description: The query to partially evaluate and compile.
+    CompileOptions:
+      type: object
+      description: Additional options to use during partial evaluation. Only the disableInlining option is currently supported in OPA. Enterprise OPA may support additional options.
+      additionalProperties: true
+      properties:
+        disableInlining:
+          type: array
+        targetDialects:
+          type: array
+          description: The output targets for partial evaluation. Different targets will have different constraints.
+          items:
+            type: string
+            enum:
+              - ucast+all
+              - ucast+minimal
+              - ucast+prisma
+              - ucast+linq
+              - sql+mssql
+              - sql+mysql
+              - sql+postgres
+        targetSQLTableMappings:
+          type: object
+    CompileUnknowns:
+      type: array
+      description: The terms to treat as unknown during partial evaluation.
+      items:
+        type: string
+      default: ["input"]
+    CompileResultJSON:
+      type: object
+      description: The partially evaluated result of the query. Result will be empty if the query is never true.
+      properties:
+        result:
+          type: object
+          description: The partially evaluated result of the query.
+          additionalProperties: true
+          properties:
+            queries:
+              type: array
+            support:
+              type: object
+    CompileResultUCAST:
+      type: object
+      description: The partially evaluated result of the query, in UCAST format. Result will be empty if the query is never true.
+      properties:
+        result:
+          type: object
+          description: The partially evaluated result of the query as UCAST.
+          additionalProperties: true
+          properties:
+            queries:
+              type: array
+              description: Array of UCAST JSON objects describing each query.
+    CompileResultSQL:
+      type: object
+      description: The partially evaluated result of the query, in SQL format. Result will be empty if the query is never true.
+      properties:
+        result:
+          type: object
+          description: The partially evaluated result of the query as SQL.
+          additionalProperties: true
+          properties:
+            queries:
```

```suggestion
            query:
```

But to be aligned with below, I think I'd rather drop "query"/"queries" altogether, except for where the legacy Compile API demands it (plain AST). So SQL would become
```json
{ "result": "WHERE ticket.tenant = 3" }
```
etc

---

Turns out I've changed my mind. We should _keep_ `result` wrapping, and we should also not have the result be a un-named string. If we, in the future, wanted to return parameterized query + vars, or some extra metadata, we'll be annoyed by ourselves for having changed this to a naked string.

But I still insist on "query" (singular) 😎 Because we're returning a single OR-of-ANDs query.